File: PUGD-frontend/graphql/queries/acquisition/order.js
import gql from "graphql-tag";

const GetOrders = gql`
  query($type: String!) {
    getOrders(type: $type) {
      _id
      status
      establishement
      date
      provider
      delivery_address
      billing_address
      order_lines
    }
  }
`;

const GetOrder = gql`
  query($id: String!) {
    getOrder(id: $id) {
      status
      establishement
      financial_year
      quotation_number
      order_number
      notes
      date
      provider
      type
      delivery_address
      billing_address
      order_lines
      name
    }
  }
`;

module.exports = {
  GetOrder,
  GetOrders,
};

File: PUGD-api/graphql/types/AcquisitionTypes/order.go
package AcquisitionTypes

import "github.com/graphql-go/graphql"

var OrderType = graphql.NewObject(graphql.ObjectConfig{
	Name: "OrderType",
	Fields: graphql.Fields{
		"_id": &graphql.Field{
			Type: graphql.ID,
		},
		"type": &graphql.Field{
			Type: graphql.String,
		},
		"quotation_number": &graphql.Field{
			Type: graphql.String,
		},
		"order_number": &graphql.Field{
			Type: graphql.String,
		},
		"status": &graphql.Field{
			Type: graphql.String,
		},
		"name": &graphql.Field{
			Type: graphql.String,
		},
		"establishement": &graphql.Field{
			Type: graphql.String,
		},
		"provider": &graphql.Field{
			Type: graphql.String,
		},
		"financial_year": &graphql.Field{
			Type: graphql.String,
		},
		"date": &graphql.Field{
			Type: graphql.String,
		},
		"currency": &graphql.Field{
			Type: graphql.String,
		},
		"delivery_address": &graphql.Field{
			Type: graphql.String,
		},
		"billing_address": &graphql.Field{
			Type: graphql.String,
		},
		"notes": &graphql.Field{
			Type: graphql.String,
		},
		"order_lines": &graphql.Field{
			Type: graphql.NewList(graphql.ID),
		},
		"orders": &graphql.Field{
			Type: graphql.NewList(graphql.ID),
		},
		"received": &graphql.Field{
			Type: graphql.Boolean,
		},
		"factured": &graphql.Field{
			Type: graphql.Boolean,
		},
	},
})
